src/libsync/creds/credentialmanager.cpp

```diff
@@ -73,24 +74,15 @@ QKeychain::Job *CredentialManager::set(const QString &key, const QVariant &data)
     timer->setInterval(tiemoutC);
     timer->setSingleShot(true);
 
-    auto timedOut = std::make_unique<bool>(false);
-    connect(timer, &QTimer::timeout, writeJob, [writeJob, timedOut = timedOut.get()] {
-        *timedOut = true;
-        Q_EMIT writeJob->finished(writeJob);
-        writeJob->deleteLater();
-    });
-    connect(writeJob, &QKeychain::WritePasswordJob::finished, this, [writeJob, timer, key, timedOut = std::move(timedOut), this] {
-        timer->stop();
+    Utility::ChronoElapsedTimer elapsedTimer;
+    connect(timer, &QTimer::timeout, writeJob, [writeJob] { qCWarning(lcCredentialsManager) << "set" << writeJob->key() << "did not finish in 5s!!!"; });
```

Comment:
```suggestion
    connect(timer, &QTimer::timeout, writeJob, [writeJob] { qCWarning(lcCredentialsManager) << "set" << writeJob->key() << "has not finished yet in an appropriate amount of time"; });
```

We may want to change the value later on. I could also imagine restarting the timer a couple of more times to log additional intervals, e.g., 5s, 10s, 20s, 30s.